### ISSUE TITLE
Reader View on Safari, Mac, iPhone, iPad

### ISSUE THREAD

**Issue by randycharach** [elementor/elementor#11118]
Safari for Mac, also used on iPhones and iPad, has a feature called Reader View. 
The Text is presented in an easy to read format, font and size.
Elementor does not work with reader view.
Adding this feature would make Elementor more suitable for content sites with a lot of articles. 
Reder view is commonly used among the many Apple users and available on most sites not built with Elementor. It is an important feature.

**Comment by ianstudio** [comment]
+1

**Comment by genowales** [comment]
+1

**Comment by daniilicic** [comment]
+1

**Comment by jttmckee** [comment]
And I thought I was doing something wrong!  This is a serious bug.  I've built a website from scratch before and Safari Reader View just worked (like the other reader views).  Elementor must be breaking something.  Please fix this.

**Comment by adamconnell** [comment]
Damn, this is a pretty significant bug. Seriously needs fixing!

**Comment by randycharach** [comment]
Someone else brought this up in 2018. https://github.com/elementor/elementor/issues/5639 

Now that they've recently received an additional 15M in funding, perhaps they'll use a fraction of those funds to fix this major issue. 

Based on their 2018 response, as well as what they told me, it's going to take a bit of noise to make this a priority. 

I regret using Elememtor to build my site for this reason and shocked that 4M users haven't really noticed this deficiency. 

Bottom line -  spread the news - and get more comments and +1s on this thread.

**Comment by mthompsontpm** [comment]
+1

**Comment by wesleygoldsborough** [comment]
+1 Would like to see this too

**Comment by mellymine** [comment]
+1  this could be a dealbreaker in my being able to use Elementor for future projects :-(

**Comment by randycharach** [comment]
I am no longer using Elementor. Rebuilt my sites with Generatepress and will never look back.

**Comment by thomas-kuehn** [comment]
+1, not acceptable in 2020. 
Please fix this bug soon.

**Comment by elaine-fabrica** [comment]
Hi all,

It's 2021 and I'm having the same issue but noticed I do get the H1 and H2 showing most times.
However, I did randomly get one article displaying correctly on the reader's view on desktop, but not on mobile.
https://fabricacollective.com/practical-psychology/

I'll ask on the facebook community what's the status on that issue.

**Comment by d19dotca** [comment]
It's disappointing to see this issue is still present. Not only for Safari but apps like Twitter can have Reader mode enabled by default when clicking links to website too (this is how it is for my Twitter app config), so it's not solely affecting people using Safari only, but the Twitter iOS app too in my experience and there may likely be other similar app experiences. 

I sure hope this issue will be fixed. I had actually reported it as a bug to Apple last year in the beta reporting app thinking it was their issue, but now seeing this I suspect this is an incompatibility issue from the Elementor side of the code.

**Comment by careyjames** [comment]
im out, fired.

**Comment by SEwebm** [comment]
2023, issue still persists.

As far as I understand Safari's Implementation of the reader view / mode the reason why this is happening probably has to do with the html structure of elementor pages. 
There are multiple nesting layers of non semantic wrapper elements, combined with missing aria attributes.

Also used libraries like swiper are not optimized for accessibility, like at all. Even if the reader mode works, slides are showing multiple times because of the duplicated slides and stuff like that...

But elementor is not entirely to blame here, no one really knows how the Safari reader works exactly. Many hours of research have only led me to a 10+ years old official javascript that at least used to be part of Safari.

The "container" approach could help, I haven't tested this feature yet.

That being said, I hope there will be a solution to this problem in the future.

**Comment by Vegworks** [comment]
I am going through the same problem and I would like to find a solution. Hints or html tags like <br>, <p>, etc. or even a js-snipet to embed in the Customiser/ Custom CSS/JS “root” field is welcome.

**Comment by mrsherman28** [comment]
+1 - This has been an issue since we've converted our website to Wordpress two years ago. Elementor is a fantastic tool but this is one feature it has not been able to provide. We have many users who want the ability to use reader view on their iPads to adjust text size/font and it would also help with accessibility. Wordpress native blog posts have no problem with displaying reader view, so it is clearly a problem with Elementor and how they structure pages. Please fix this soon!

**Comment by frysteen** [comment]
Had same issue - which lead me here... and a bunch of other places.

Turns out, just needed to set the container (where the main content is) HTML Tag to be "main" instead of "default" ... and all started working.

![Screenshot 2023-04-27 at 4 52 32 pm](https://user-images.githubusercontent.com/7022544/234782983-99bce617-f8fb-408a-9dcc-f4b56eca50ce.jpg)

**Comment by combatlobsang** [comment]
I sent the issue (13 december 2024) to Elementor technical department and this is the answer:

_Please remember that it is related to a known issue. Our developers are aware of this and are working on it. A full report has been created. This means it is already in the process of resolving this issue. Therefore, the fix should be available in some of the following updates. I can not provide any ETA for this since it is not mentioned in the report, but for your further reference, it is recorded in our internal bugs database under BFJSM-236. There is no known workaround at this point, this feature will need a code update from our end.We are doing our best to have a stable and well-tested product as much as possible. Glitches like this simply happen, and we are terribly sorry for this and for any inconvenience you may have because of this._ 

In see websites as Vilma Nuñez's build with Elementor but with Read Mode in Safari ok:
https://vilmanunez.com/conoce-las-campanas-de-black-friday-de-target-y-walmart/

Any ideas?

**Comment by hirschferkel** [comment]
Changing HTML Tag to "main" did not work for my page. Only 1/3 of the content is available in "Readers View".

**Comment by hirschferkel** [comment]
@combatlobsang I figured out, why this issue happens at least in my case. It depends if there is an additional HTML-element in Elementor Text Widget. E.g. when you apply a different color in headline styling, the render view already breaks.

In my case, I have a grey standard text and render view is working in Safari: [Render View Working Link](https://test.ixtract.de/test_reader_view_working/)

As soon as I apply a black color e.g. to the headline, the render view breaks:[Render View Not Working Link](https://test.ixtract.de/test_reader_view_not_working/)

![Bildschirmfoto 2024-02-04 um 15 46 23](https://github.com/elementor/elementor/assets/16016422/c22342b0-587e-4937-80f9-d0e505f75b03)
![Bildschirmfoto 2024-02-04 um 15 46 12](https://github.com/elementor/elementor/assets/16016422/477704d0-5673-498d-998d-616202195e7e)

**Comment by frysteen** [comment]
@combatlobsang - neither show in reader on my iPhone. 

Safari is just being silly. Have tried all sorts of tags (main, div, article) and nothing is consistent in what shows. 

I have exact same post types, with exact same layout and html, some show reader properly, some don’t. 

Some don’t show the reader icon but still work if you manually enable reader.

**Comment by bentographics** [comment]
+1 still broken. Changing the HTML tag to "main" didn't work either.

**Comment by cspeloso** [comment]
+1 still not working. It looks like it's grabbing the first H2 on my page and not showing anything past then. I'm assuming it would work if I hard coded the HTML in like <h2>Subtitle</h2><p>more content</p>, but that would defeat the point of elementor....

how has this not been solved yet...
